### PR TITLE
Dockerfile.e2e: copy test sources

### PR DESCRIPTION
**Pull request by kolyshkin**
(this is a quick followup to #39116)

Package "gotest.tools/assert" uses source introspection to
print more info in case of assertion failure. When source code
is not available, it prints an error instead.

In other words, before this commit:

> --- SKIP: TestCgroupDriverSystemdMemoryLimit (0.00s)
>     cgroupdriver_systemd_test.go:32: failed to parse source file: /go/src/github.com/docker/docker/integration/system/cgroupdriver_systemd_test.go: open /go/src/github.com/docker/docker/integration/system/cgroupdriver_systemd_test.go: no such file or directory
>     cgroupdriver_systemd_test.go:32:

and after:

> --- SKIP: TestCgroupDriverSystemdMemoryLimit (0.09s)
>    cgroupdriver_systemd_test.go:32: !hasSystemd()

This increases the resulting image size by about 2 MB
on my system (from 758 to 760 MB).

### CODE REVIEW

File: Dockerfile.e2e [hunk 1]
```diff
@@ -43,8 +43,11 @@ FROM base AS builder
 ARG DOCKER_GITCOMMIT
 ENV DOCKER_GITCOMMIT=${DOCKER_GITCOMMIT:-undefined}
 COPY . .
+# Copy test sources tests that use assert can print errors
+RUN mkdir -p /build/${PWD} && find integration integration-cli -name \*_test.go -exec cp --parents '{}' /build${PWD} \;
```
Comment (kolyshkin) [inline Dockerfile.e2e]: I'm using $PWD here as it points to /go/src/github.com/docker/docker and this is where binaries expect the files to be in. This is also why the last COPY statement (below) has destination as /

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Left some comments, but otherwise looks good 🤗

File: Dockerfile.e2e [hunk 2]
```diff
@@ -43,8 +43,11 @@ FROM base AS builder
 ARG DOCKER_GITCOMMIT
```
[the remark this answers is not in the document]
Comment (kolyshkin) [reply Dockerfile.e2e]: DONE

**Comment by kolyshkin** [comment]
@seemethere PTAL (second patch)

**Review by seemethere — APPROVED** [review]

File: Dockerfile.e2e [hunk 3]
```diff
@@ -40,11 +40,13 @@ RUN CGO_ENABLED=0 go build -buildmode=pie -o /build/httpserver/httpserver github
 FROM base AS builder
 
 # Set tag and add sources
-ARG DOCKER_GITCOMMIT
-ENV DOCKER_GITCOMMIT=${DOCKER_GITCOMMIT:-undefined}
 COPY . .
+# Copy test sources tests that use assert can print errors
+RUN mkdir -p /build${PWD} && find integration integration-cli -name \*_test.go -exec cp --parents '{}' /build${PWD} \;
+# Build and install test binaries
+ARG DOCKER_GITCOMMIT=${DOCKER_GITCOMMIT:-undefined}
```
Comment (thaJeztah) [inline Dockerfile.e2e]: This should do the job;

```suggestion
ARG DOCKER_GITCOMMIT=undefined
```

Comment (kolyshkin) [reply Dockerfile.e2e]: Right; fixed

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by seemethere — APPROVED** [review]

**Pull request merged**